### PR TITLE
Add support for http PATCH method to RestClient

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

**Review by petrdvorak — APPROVED** [review]
Looks good

**Review by saalistaja — APPROVED** [review]

**Review by romanstrobl — CHANGES_REQUESTED** [review]
Can you please update the documentation, too? Thanks.

File: README.md [hunk 1]
```diff
@@ -156,7 +156,10 @@ Once the rest client is initialized, you can use the following methods. Each met
 - `postObject` - a blocking POST call with `ObjectRequest` / `ObjectResponse`
 - `put` - a blocking PUT call with a generic request / response
 - `putNonBlocking` - a non-blocking PUT call with a generic request / response with `onSuccess` and `onError` consumers
-- `putObject` - a blocking PUT call with `ObjectRequest` / `ObjectResponse`
+- `putObject` - a blocking PATCH call with `ObjectRequest` / `ObjectResponse`
+- `patch` - a blocking PATCH call with a generic request / response
+- `patchNonBlocking` - a non-blocking PUT call with a generic request / response with `onSuccess` and `onError` consumers
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply README.md]: Ups, wrong find/replace, fixed.

**Review by romanstrobl — APPROVED** [review]
Approved.

**Pull request merged**